### PR TITLE
Start implementing viewports_update_position

### PR DESCRIPTION
**Pull request by marijnvdwerf**
I want to put `viewport_target_sprite`, `saved_view_y`, and `saved_view_x` into a sub struct, so we can just loop through viewports.

### CODE REVIEW

File: src/openloco/window.cpp [hunk 1]
```diff
@@ -18,6 +21,236 @@ namespace openloco::ui
...
+            }
+        }
+
+        if (this->viewports[1] != nullptr)
+        {
+            this->call_on_resize();
+
+            viewport* viewport = this->viewports[0];
+            if (viewport != nullptr)
+            {
```
Comment (IntelOrca) [inline src/openloco/window.cpp]: Should there be code in this block?

Comment (marijnvdwerf) [reply src/openloco/window.cpp]: It's what the assembly does; first a null check, then call resize and then another null check. Though I guess the first one should be pretty safe to remove

File: src/openloco/window.cpp [hunk 2]
```diff
@@ -18,6 +21,236 @@ namespace openloco::ui
...
+        return regs.edx;
+    }
+
+    void sub_4CA444(int16_t x, int16_t y, int16_t z, int16_t* outX, int16_t* outY)
+    {
+        registers regs;
+        regs.ax = x;
+        regs.bx = y;
+        regs.cx = z;
+        call(0x00467297, regs);
```
Comment (AaronVanGeffen) [inline src/openloco/window.cpp]: Should be `0x004CA444`

File: src/openloco/window.cpp [hunk 3]
```diff
@@ -33,6 +38,169 @@ namespace openloco::ui
...
+        *outX = regs.ax;
+        *outY = regs.bx;
+    }
+
+    // 0x004C6456
+    void window::viewports_update_position()
+    {
+        this->call_resize();
+
+        for (int i = 0; i < 2; i++)
```
Comment (marijnvdwerf) [inline src/openloco/window.cpp]: Maybe introduce constant for viewport count?

**Pull request merged**